pygccxml/parser/source_reader.py

```diff
@@ -159,28 +160,8 @@ def __create_command_line_castxml(self, source_file, xmlfile):
             # On mac or linux, use gcc or clang (the flag is the same)
             cmd.append('--castxml-cc-gnu ')
 
-            # Check for -std=xx flags passed to the compiler.
-            # A regex could be used but this is a moving target.
-            # See c++1z for example. It is preferable to have a defined
-            # list of what is allowed. http://clang.llvm.org/cxx_status.html
-            #
-            # Version 98 and 03 are only there in the case somebody is using
-            # these flags; this is the equivalent to not passing these flags.
-            standards = [
-                "-std=c++98",
-                "-std=c++03",
-                "-std=c++11",
-                "-std=c++14",
-                "-std=c++1z"]
-
-            std_flag = ""
-            for standard in standards:
-                if standard in self.__config.cflags:
-                    std_flag = " " + standard + " "
-
-            # A -std= flag was passed, but is not in the list
-            if "-std=" in self.__config.cflags and std_flag == "":
-                raise(RuntimeError("Unknown -std=c++xx flag used !"))
+            std_flag = "" if self.__cxx_std.is_implicit \
+                else " " + self.__cxx_std.stdcxx + " "
```

Comment:
Could you rewrite this line without using a ternary operator ? I think there are no ternary operators in the code (or very few), so lets keep the style consistent. I also find if/else structures more readable; (though longer :))